### ISSUE TITLE
vimPlugins.neogit: broken on 24.05

### ISSUE THREAD

**Issue by t-gebauer** [NixOS/nixpkgs#318845]
### Describe the bug
The current version of `vimPlugins.neogit` (master@2024-05-16) does not work in combination with the current version of neovim (0.9.5) in 24.05.
`neogit` uses some nvim 0.10 APIs.

See: https://github.com/NeogitOrg/neogit/issues/1312#issuecomment-2122502161

### Steps To Reproduce
1. Install `neovim` with packages `neogit` and `plenary-nvim`
2. `:Neogit`

### Expected behavior
No errors. Git status is shown (when in a git repository).

### Additional Context
As mentioned in the linked issue: `v0.0.1` works

```nix
  _neogit = pkgs.vimUtils.buildVimPlugin {
    name = "neogit";
    src = pkgs.fetchFromGitHub {
      owner = "NeogitOrg";
      repo = "neogit";
      rev = "v0.0.1"; # neovim 0.9.5 compatible
      hash = "sha256-xncUButV2/FSOPNolbrZuPPSC1FOtgrlZBwzHHPKCTo=";
    };
  };
```

### Notify maintainers

@GaetanLepage
@teto

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by teto** [comment]
Would you be willing to submit a PR please ? One way to prevent this is I am pushing for running more of the plugin tests (neogit tests are not run typically, which would have caught this most likely).

**Comment by t-gebauer** [comment]
No, sorry. No time right now to figure out what needs to be done for a PR. Maybe next weekend.

**Comment by GaetanLepage** [comment]
The plugins have been updated since then. Is the issue still occuring @t-gebauer ?

**Comment by t-gebauer** [comment]
I can't really check right now. But according to the versions the problem should be unchanged.
On 24.05 neovim is at 0.9.5 and neogit tracks its master branch -> incompatible.
On master neovim is already on 0.10.0 and neogit master should therefore work..

**Comment by GaetanLepage** [comment]
Oh, sorry, I thought for a moment that the issue was on master.
For me, this is a more general issue. Neovim is 0.9.5 on nixos-24.05 and is not planned to be updated to 0.10.0.
So, plugin updates should not be backported to the stable channel. Actually, I don't think that they are.

If plugins are indeed frozen on 24.05, why not manually downgrade `vimPlugins.neogit` manually on 24.05 to ensure compatibility with nvim 0.9.5 ?

**Comment by teto** [comment]
a PR to downgrade neogit on the 24.05 branch https://github.com/NixOS/nixpkgs/tree/nixos-24.05 will fix it.

**Comment by GaetanLepage** [comment]
Fixed by #320028
